### PR TITLE
MGMT-13089: is_update_needed should return true if the cluster logs_i…

### PR DESCRIPTION
**Pull request by eranco74**
…nfo isn't completed or timedout

### CODE REVIEW

**Comment by osherdp** [comment]
/hold
need to manually check the effects on the gather step

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 1]
```diff
@@ -295,6 +301,9 @@ def get_cluster_events_path(cluster, output_folder):
     return os.path.join(output_folder, f"cluster_{cluster['id']}_events.json")
 
 
+def get_cluster_metadata(output_folder):
+    return os.path.join(output_folder, f"metadata.json")
```
Comment (osherdp) [inline src/assisted_test_infra/download_logs/download_logs.py:305]: a couple of suggestions embedded in one:
```suggestion
def get_cluster_metadata(output_folder: str) -> dict:
    with open(os.path.join(output_folder, "metadata.json"), "rt") as f:
        return json.load(f)
```

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 2]
```diff
@@ -106,6 +106,12 @@ def is_update_needed(output_folder: str, update_on_events_update: bool, client:
             os.remove(destination_event_file_path)
             latest_event_tp.close()
             need_update = True
+    # if the logs info in the cluster metadata isn't set to final status we should update
+    cluster_md = get_cluster_metadata(output_folder)
+    with open (cluster_md) as f:
+        md = json.load(f)
+        if md.get("logs_info", "") not in ("timeout", "completed", ""):
+            need_update = True
```
Comment (osherdp) [inline src/assisted_test_infra/download_logs/download_logs.py:114]: nit: add a blank line for having a breath when reading the code :)

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 1 above)
Comment (eranco74) [reply src/assisted_test_infra/download_logs/download_logs.py:305]: I would but similar functions in this file return file path...

Comment (eranco74) [reply src/assisted_test_infra/download_logs/download_logs.py]: NM, fixed

File: src/assisted_test_infra/download_logs/download_logs.py (hunk 2 above)
Comment (eranco74) [reply src/assisted_test_infra/download_logs/download_logs.py]: Now that the open and json.load moved elsewhere it should be easier to read.

**Review by osherdp — APPROVED** [review]

**Comment by eranco74** [comment]
> /hold need to manually check the effects on the gather step

How do I do that?

**Comment by osherdp** [comment]
looking at https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-test-infra/1974/pull-ci-openshift-assisted-test-infra-master-e2e-metal-assisted/1607723037539438592/artifacts/e2e-metal-assisted/assisted-common-gather/artifacts/2022-12-27_13-33-26_730af25c-333f-401d-83c1-fd6bc5e09f2a/ and seeing that all files are in there
for me it looks good

File: src/assisted_test_infra/download_logs/download_logs.py [hunk 3]
```diff
@@ -89,6 +89,12 @@ def is_update_needed(output_folder: str, update_on_events_update: bool, client:
     if not os.path.isdir(output_folder):
         return True
 
+    # if the logs info in the cluster metadata isn't set to final status we should update
+    cluster_md = get_cluster_metadata(output_folder)
+    if cluster_md.get("logs_info", "") not in ("timeout", "completed", ""):
+        log.info(f"update needed, missing must-gather logs")
```
Comment (osherdp) [inline src/assisted_test_infra/download_logs/download_logs.py:95]:
```suggestion
        log.info("update needed, missing must-gather logs")
```

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by eranco74** [comment]
Works, verfied by this job:
http://assisted-jenkins.usersys.redhat.com/job/download_logs/45765/console
We can see that it added the logs to this ticket https://issues.redhat.com/browse/AITRIAGE-4840

**Review by osherdp — APPROVED** [review]

**Pull request merged**